### PR TITLE
fix: DROP stream for persistent query doesn't always drop underlying query

### PR DESCRIPTION
**Pull request by spena**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_
This PR fixes an issue when terminating an INSERT query that caused the underlying CREATE_AS query to not be terminated when dropping the stream.

i.e.
```
ksql> create stream p1 as select * from s1;
ksql> insert into p1 select * from s2;
ksql> show queries;
Query ID       | Query Type | Status    | Sink Name | Sink Kafka Topic | Query String                                                                                            
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 INSERTQUERY_43 | PERSISTENT | RUNNING:1 | P1        | P1               | insert into p1 select * from s2;                                                                        
 CSAS_P1_41     | PERSISTENT | RUNNING:1 | P1        | P1               | CREATE STREAM P1 WITH (KAFKA_TOPIC='P1', PARTITIONS=1, REPLICAS=1) AS SELECT * FROM S1 S1 EMIT CHANGES; 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
ksql> terminate INSERTQUERY_43;
ksql> drop stream p1;
ksql> show queries;
Query ID       | Query Type | Status    | Sink Name | Sink Kafka Topic | Query String                                                                                            
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CSAS_P1_41     | PERSISTENT | RUNNING:1 | P1        | P1               | CREATE STREAM P1 WITH (KAFKA_TOPIC='P1', PARTITIONS=1, REPLICAS=1) AS SELECT * FROM S1 S1 EMIT CHANGES; 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

The `CSAS_P1_41` should have been terminated automatically with DROP stream.

The problem was in `QueryRegistryImpl` that was removing query references from both `insertQueries` and `createAsQueries` mapping variables. This PR now removes the references from the right map variable.

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryRegistryImplTest.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import io.confluent.ksql.query.QueryRegistryImpl.QueryExecutorFactory;
 import io.confluent.ksql.schema.ksql.LogicalSchema;
 import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.util.KsqlConstants;
```
Comment (wcarlson5) [inline ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryRegistryImplTest.java:27]: Can we test the query gets removed from only the correct list here? It seems that would cover the case that was causing the the problem

**Review by wcarlson5 — COMMENTED** [review]
This makes sense to me, I just got a couple questions about the testing 👍

File: ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java [hunk 2]
```diff
@@ -290,16 +294,21 @@ private void unregisterQuery(final QueryMetadata query) {
       final QueryId queryId = persistentQuery.getQueryId();
       persistentQueries.remove(queryId);
 
-      // If query is a INSERT query, then this line should not cause any effect
```
[the remark this answers is not in the document]
Comment (spena) [reply ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java:293]: Correct. I thought it was true when I implemented this feature. I don't know why it wasn't it.

File: ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryExecutorTest.java [hunk 3]
```diff
@@ -52,6 +52,7 @@
 import io.confluent.ksql.services.KafkaTopicClient;
 import io.confluent.ksql.services.ServiceContext;
 import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlConstants;
```
[the remark this answers is not in the document]
Comment (spena) [reply ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryExecutorTest.java:55]: Done

File: ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryRegistryImplTest.java (hunk 1 above)
Comment (spena) [reply ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryRegistryImplTest.java:28]: Done

**Review by wcarlson5 — APPROVED** [review]
LGTM thanks for the extra tests @spena !

**Comment by guozhangwang** [comment]
Thanks @spena . The patch LGTM.

I have an out-of-scope question following this though: back to your example in the description:

```
create stream p1 as select * from s1;
insert into p1 select * from s2;
```

if we call `drop s1; drop s2;` would that cause `p1` to be automatically dropped as well?

**Comment by spena** [comment]
@guozhangwang 

> create stream p1 as select * from s1;
insert into p1 select * from s2;
if we call drop s1; drop s2; would that cause p1 to be automatically dropped as well?

The DROP commands won't work because another stream/table is reading/writing from S1 or S2. There is a constraint linked from p1 -> s1 as well as p1 -> s2. In order to drop s1 and s2, you must first drop the p1 query and/or terminate the insert query.

**Comment by guozhangwang** [comment]
Thanks!

> In order to drop s1 and s2, you must first drop the p1 query and/or terminate the insert query.

When we drop p1, wouldn't both queries be automatically dropped too? Only s1/s2 would still exist.

**Comment by spena** [comment]
@guozhangwang 

> When we drop p1, wouldn't both queries be automatically dropped too? Only s1/s2 would still exist.

No. That was proposed in the KLIP, but it had many discussions. At the end we implemented the klip partially to at least drop a stream+query if there was no other constraints.

**Pull request merged**